### DISCUSSION TITLE
Calculation based on ELR Method

### DISCUSSION THREAD

**Discussion by muhammadshamsherkhan2002** [casact/chainladder-python#247] [Q&A]
I hope that you are fine and perfect.

I would like to ask you that is there any availability of calculation based on the **ELR Method** while using chainladder-python library? If yes, please guide me (with example code in library) further that it will support the following feature in ELR model:

**Either**
1.Single ELR rate (to be applied on all origin periods).
**Or**
2.Different ELR rates (to be applied origin period wise- where needed).

Please go through the attached sample excel file (just for understanding purpose). There are two sheets in the excel file.
- calc. by single ELR rate
- calc. by different ELR rate (period wise - where needed)

Your kind response is awaited in this regard.

[ELR Method - Sample Working Sheet.xlsx](https://github.com/casact/chainladder-python/files/7902524/ELR.Method.-.Sample.Working.Sheet.xlsx)

**Comment by kennethshsu** [comment]
I don't think this currently exists. 

I can work on this.

**Reply by muhammadshamsherkhan2002** [reply]
Could you provide this method in python-chailadder library? We will be very grateful to you.

- The Formula for the ELR Method is
   ELR Method=EP ∗ ELR −Paid Losses
   where:
   EP = Earned premiums
​
This is just for your reference purpose. (please find again attached excel file below with correction)

[ELR Method - Sample Working Sheet.xlsx](https://github.com/casact/chainladder-python/files/7902965/ELR.Method.-.Sample.Working.Sheet.xlsx)

**Comment by jbogaardt** [comment]
The ELR method is can be achieved with `cl.Benktander(n_iters=0, apriori=0.6)`. It is described in the docs [here](https://chainladder-python.readthedocs.io/en/latest/methods.html#benktander).

Enough people have not recognized this relationship between Benktander and ELR that maybe it is worth making a dedicated estimator that hands off work to Benktander with `n_iters=0`. The `BornhetterFerguson` method is written this way, but with ` n_iters=1`.

**Reply by kennethshsu** [reply]
I'll make a note of this in the tutorial.

Looking at the documentation, the 2nd piece of the equation (summation), how does summing from 0 to -1 work? I get that the sum should result in 0. Is there a way to be clearer with the formula?

And I agree that we should add another estimator and just use call the Benktander's more general method. If we have one for BF, we should have one for expected loss method. Let me try to do this.

**Reply by jbogaardt** [reply]
> how does summing from 0 to -1 work?

I think the equation is suggesting that you should sum from k=0 to k=n-1 where n is the number of iterations chosen in the Benktander method. I'm sure I pulled it out of a paper somewhere. Should probably have cited it.  


We should really think about what should go into the tutorials and what should go into the user guide and adhere to it. Without clear guidelines we have a lot of stuff in both.  As you know, my thoughts are tutorials should be lighter on details but richer on end-to-end use-cases. User Guide is rich on details but light on end-to-end use-cases.

**Reply by jbogaardt** [reply]
@kennethshsu, I realize now you mean in the special case of n_iters=0. I don't quite understand the formula either, but the math works in the implementation - maybe I wrote the formula wrong? 🤷‍♂️

**Reply by kennethshsu** [reply]
I actually think it does sum to 0, according to [wolfram](https://www.wolframalpha.com/input/?i2d=true&i=sum+n%5C%2844%29+n%3D0+to+-1). 😅
